trafaret/base.py

```diff
@@ -1246,36 +1270,39 @@ def __repr__(self):
         return "<Mapping(%r => %r)>" % (self.key, self.value)
 
 
-class Enum(Trafaret):
+class Enum(Literal):
     """
-    >>> trafaret = Enum("foo", "bar", 1) >> ignore
-    >>> trafaret
-    <Enum('foo', 'bar', 1)>
-    >>> trafaret.check("foo")
-    >>> trafaret.check(1)
-    >>> extract_error(trafaret, 2)
-    "value doesn't match any variant"
-    >>> trafaret.is_valid(1)
-    True
-    >>> trafaret.is_valid(2)
-    False
+    Deprecated: use Literal.
     """
-    __slots__ = ['variants']
+    def __init__(self, *args, **kwargs):
+        deprecated("Use Literal.")
+        super().__init__(*args, **kwargs)
 
-    def __init__(self, *variants):
-        self.variants = variants[:]
 
-    def check_value(self, value):
-        if value not in self.variants:
+class ToEnum(Trafaret):
+    """
+    Converts value to an Enum instance.
+
+    >>> class MyEnum(enum.Enum):
+    ...     foo = 3
+    >>> ToEnum(MyEnum).check(3)
+    <MyEnum.foo: 3>
+    """
+    __slots__ = ["enum"]
+
+    def __init__(self, enum_: enum.Enum):
+        self.enum = enum_
+
+    def check_and_return(self, value) -> enum.Enum:
+        try:
+            return self.enum(value)
+        except ValueError:
             self._failure(
-                "value doesn't match any variant",
+                "not a valid value for %s" % self.enum,
                 value=value,
-                code=codes.DOES_NOT_MATCH_ANY,
+                code=codes.NOT_ENUM,
```

Comment:
Is not this error code misleading a bit? Maybe it should be more like `not_member_of_enum`?